pkg/controllermanager/controller/backupinfrastructure/backup_infrastructure_control.go

```diff
@@ -190,7 +191,8 @@ func (c *defaultControl) ReconcileBackupInfrastructure(obj *gardenv1beta1.Backup
 	// When this happens the controller will remove the finalizer from the BackupInfrastructure so that it can be garbage collected.
 	if backupInfrastructure.DeletionTimestamp != nil {
 		gracePeriod := computeGracePeriod(c.config)
-		if time.Now().Sub(backupInfrastructure.DeletionTimestamp.Time) > gracePeriod {
+		present, _ := strconv.ParseBool(backupInfrastructure.ObjectMeta.Annotations[common.BackupInfrastructureForceDeletion])
```

Comment:
Same here - error check needed?!

Reply:
At this stage, we can safely ignore it. Since the erroneous case is supposed to be handled at admission/validation phase.